### ISSUE TITLE
fiberassign is only processing one out of six dark tiles

### ISSUE THREAD

**Issue by moustakas** [desihub/fiberassign#125]
I'm running fiber assignment with a non-standard tile file and only one out of six input tiles are being processed by fiber assignment.  I'm certain I'm doing something wrong (probably with my input tile file) but I'm not sure what to change.  

To reproduce the issue you'll need to unpack this tarball in any empty directory:
http://www.sos.siena.edu/~jmoustakas/tmp/test-fiberassign.tar.gz
and then run
```
fiberassign  --mtl mtl.fits --stdstar standards-dark.fits --sky sky.fits \
  --surveytiles dark-tiles.txt --footprint tiles.fits \
  --fibstatusfile fiberstatus.ecsv --outdir .
```

There are six input tiles, tile numbers `10000-100005` and everything runs successfully.  However, the only resulting output file is `'tile_10000.fits`. 

@sbailey @forero @rncahn your thoughts would be welcome.

Here's the full log:
```
# Read target, SS, SF files at 6.7e-05 s
star mask 60129542144
Finding file: standards-dark.fits
Found MTL input file: standards-dark.fits
Reading MTL input file standards-dark.fits
NUMOBS_MORE not found ... setting to 1
PRIORITY not found ... setting to 1
Keeping 2404 targets within ra/dec ranges
star mask 0
Finding file: sky.fits
Found MTL input file: sky.fits
Reading MTL input file sky.fits
NUMOBS_MORE not found ... setting to 1
PRIORITY not found ... setting to 1
Keeping 94208 targets within ra/dec ranges
star mask 0
Finding file: mtl.fits
Found MTL input file: mtl.fits
Reading MTL input file mtl.fits
Keeping 19217 targets within ra/dec ranges
# ...read targets  took : 0.08 s
 Target size 19217 
 Standard Star size 21621 
 Sky Fiber size 115829 
# map position in target list to immutable targetid at 0.0958 s
# assign priority classes at 0.0959 s
  class  0  number  5116
  class  1  number  3399
  class  2  number  13106
# ...priority list took : 0.0017 s
# Start positioners at 0.0976 s
before reading positioners 
read the positioner file
 sorted by fiber number
 i 0 FibPos[i].fib_num 0 
 i 1 FibPos[i].fib_num 1 
 i 2 FibPos[i].fib_num 2 
 i 3 FibPos[i].fib_num 3 
 i 4 FibPos[i].fib_num 4 
 i 5 FibPos[i].fib_num 5 
 i 6 FibPos[i].fib_num 6 
 i 7 FibPos[i].fib_num 7 
 i 8 FibPos[i].fib_num 8 
 i 9 FibPos[i].fib_num 9 
 made neighbors 
Input TimeSun Jun 21 00:00:00 2018
Current TimeSun Jun 21 00:00:00 2018
before reading status 
Read from fiber status: Fiber_pos 0 Location 95 Broken 1 Stuck 0 dates 2018-02-21T09:23:51 2100-02-21T09:24:24 
Init Time for FiberSun Feb 21 09:23:51 2018
End Time for FiberSun Feb 21 09:24:24 2100
Changing fiberastatus entry: Fiber 0 Location 95
BROKEN
Read from fiber status: Fiber_pos 1 Location 62 Broken 1 Stuck 0 dates 2018-02-21T09:23:51 2100-02-21T09:24:24 
Init Time for FiberSun Feb 21 09:23:51 2018
End Time for FiberSun Feb 21 09:24:24 2100
Changing fiberastatus entry: Fiber 1 Location 62
BROKEN
Read from fiber status: Fiber_pos 2 Location 102 Broken 0 Stuck 1 dates 2018-02-21T09:23:51 2100-02-21T09:24:24 
Init Time for FiberSun Feb 21 09:23:51 2018
End Time for FiberSun Feb 21 09:24:24 2100
Changing fiberastatus entry: Fiber 2 Location 102
STUCK
Read from fiber status: Fiber_pos 3 Location 82 Broken 0 Stuck 1 dates 2018-02-21T09:23:51 2100-02-21T09:24:24 
Init Time for FiberSun Feb 21 09:23:51 2018
End Time for FiberSun Feb 21 09:24:24 2100
Changing fiberastatus entry: Fiber 3 Location 82
STUCK
Read from fiber status: Fiber_pos 4 Location 131 Broken 0 Stuck 1 dates 2018-02-21T09:23:51 2100-02-21T09:24:24 
Init Time for FiberSun Feb 21 09:23:51 2018
End Time for FiberSun Feb 21 09:24:24 2100
Changing fiberastatus entry: Fiber 4 Location 131
STUCK
read status file
# ..posiioners  took : 0.151 s
# Start plates at 0.249 s
 number of tiles 6 
Finding file: tiles.fits
Found input tile centers file: tiles.fits
Reading input tile centers file tiles.fits
 size of P  6
# Start invert tiles at 0.00189 s
# ..inversion  took : 1.31e-05 s
# do inversion of used plates at 0.00191 s
# .. sued plates inversion  took : 0.000934 s
# Read 6 plate centers from tiles.fits and 5000 fibers from ./tmp_POS_pos.txt
# ..plates   took : 0.0032 s
# Start building HTM tree at 0.252 s
# Doing kd-tree... took : 0.0207 s
# collect galaxies at  at 0.273 s
# Begin collecting available galaxies
 # ... took : 0.165 s
# ... took : 0.165 s
# collect available tile-fibers at at 0.438 s
# Begin computing available tilefibers
# ... took : 0.0078 s
galaxies outside footprint 55277
 Nplate 6  Ngal 115829   Nfiber 5000 
# Start assignment at :  0.447 s
# Begin simple assignment :
# ... took : 0.0149 s
 countme 30000 
 Plates actually used 6 
start redistribute 
# Begin redistribute TF :
  1,042 redistributions of tile-fibers 
# ... took : 0.00441 s
# Begin improve :
 improvements  24
# ... took : 0.0264 s
start redistribute 
# Begin redistribute TF :
  231 redistributions of tile-fibers 
# ... took : 0.00402 s
# assign SS and SF  at 0.497 s
# count SS and SF  at 0.723 s
 Totals SS      0    SF   25452 class  0     904 class  1    3072 class  2       0
# print fits files  at 0.723 s
# Finished !... in 0.755 s
# Read target, SS, SF files at 5.7e-05 s
star mask 60129542144
Finding file: sky.fits
Found MTL input file: sky.fits
Reading MTL input file sky.fits
NUMOBS_MORE not found ... setting to 1
PRIORITY not found ... setting to 1
Keeping 0 targets within ra/dec ranges
star mask 0
Finding file: sky.fits
Found MTL input file: sky.fits
Reading MTL input file sky.fits
NUMOBS_MORE not found ... setting to 1
PRIORITY not found ... setting to 1
Keeping 94208 targets within ra/dec ranges
star mask 0
Finding file: sky.fits
Found MTL input file: sky.fits
Reading MTL input file sky.fits
NUMOBS_MORE not found ... setting to 1
PRIORITY not found ... setting to 1
Keeping 94208 targets within ra/dec ranges
# ...read targets  took : 0.172 s
 Target size 94208 
 Standard Star size 94208 
 Sky Fiber size 188416 
# map position in target list to immutable targetid at 0.212 s
# assign priority classes at 0.212 s
  class  0  number  94208
# ...priority list took : 0.00265 s
# Start positioners at 0.214 s
before reading positioners 
read the positioner file
 sorted by fiber number
 i 0 FibPos[i].fib_num -1 
 i 1 FibPos[i].fib_num -1 
 i 2 FibPos[i].fib_num -1 
 i 3 FibPos[i].fib_num -1 
 i 4 FibPos[i].fib_num -1 
 i 5 FibPos[i].fib_num -1 
 i 6 FibPos[i].fib_num -1 
 i 7 FibPos[i].fib_num -1 
 i 8 FibPos[i].fib_num -1 
 i 9 FibPos[i].fib_num -1 
 made neighbors 
Input TimeSun Jun 21 00:00:00 2018
Current TimeSun Jun 21 00:00:00 2018
before reading status 
Read from fiber status: Fiber_pos 0 Location 95 Broken 1 Stuck 0 dates 2018-02-21T09:23:51 2100-02-21T09:24:24 
Init Time for FiberSun Feb 21 09:23:51 2018
End Time for FiberSun Feb 21 09:24:24 2100
Read from fiber status: Fiber_pos 1 Location 62 Broken 1 Stuck 0 dates 2018-02-21T09:23:51 2100-02-21T09:24:24 
Init Time for FiberSun Feb 21 09:23:51 2018
End Time for FiberSun Feb 21 09:24:24 2100
Read from fiber status: Fiber_pos 2 Location 102 Broken 0 Stuck 1 dates 2018-02-21T09:23:51 2100-02-21T09:24:24 
Init Time for FiberSun Feb 21 09:23:51 2018
End Time for FiberSun Feb 21 09:24:24 2100
Read from fiber status: Fiber_pos 3 Location 82 Broken 0 Stuck 1 dates 2018-02-21T09:23:51 2100-02-21T09:24:24 
Init Time for FiberSun Feb 21 09:23:51 2018
End Time for FiberSun Feb 21 09:24:24 2100
Read from fiber status: Fiber_pos 4 Location 131 Broken 0 Stuck 1 dates 2018-02-21T09:23:51 2100-02-21T09:24:24 
Init Time for FiberSun Feb 21 09:23:51 2018
End Time for FiberSun Feb 21 09:24:24 2100
read status file
# ..posiioners  took : 0.00115 s
# Start plates at 0.216 s
 number of tiles 6 
Finding file: tiles.fits
Found input tile centers file: tiles.fits
Reading input tile centers file tiles.fits
 size of P  6
# Start invert tiles at 0.000224 s
# ..inversion  took : 8.82e-06 s
# do inversion of used plates at 0.00024 s
# .. sued plates inversion  took : 1.19e-05 s
# Read 6 plate centers from tiles.fits and 20 fibers from ./tmp_ETC_pos.txt
# ..plates   took : 0.000473 s
# Start building HTM tree at 0.216 s
# Doing kd-tree... took : 0.0348 s
# collect galaxies at  at 0.251 s
# Begin collecting available galaxies
 # ... took : 0.106 s
# ... took : 0.106 s
# collect available tile-fibers at at 0.357 s
# Begin computing available tilefibers
# ... took : 7.8e-05 s
galaxies outside footprint 93994
 Nplate 6  Ngal 188416   Nfiber 20 
# Start assignment at :  0.36 s
# Begin simple assignment :
# ... took : 0.000163 s
 countme 120 
 Plates actually used 6 
start redistribute 
# Begin redistribute TF :
  0 redistributions of tile-fibers 
# ... took : 2.22e-05 s
# Begin improve :
 improvements  0
# ... took : 9.06e-06 s
start redistribute 
# Begin redistribute TF :
  0 redistributions of tile-fibers 
# ... took : 1.81e-05 s
# assign SS and SF  at 0.361 s
# count SS and SF  at 0.361 s
 Totals SS      0    SF    118 class  0       0
# print fits files  at 0.361 s
# Finished !... in 0.363 s
fiberassign_exec --mtl mtl.fits  --sky sky.fits --stdstar standards-dark.fits  --fibstatusfile fiberstatus.ecsv              --outdir /var/folders/vt/6w7s8k3n5gx2nzf3jbbsqv3m0000gp/T/tmpk1t_i7e8             --surveytiles dark-tiles.txt              --footprint tiles.fits              --positioners ./tmp_POS_pos.txt             --starmask 60129542144             --rundate 2018-06-21
fiberassign_exec --mtl sky.fits  --sky sky.fits --stdstar sky.fits  --fibstatusfile fiberstatus.ecsv              --outdir /var/folders/vt/6w7s8k3n5gx2nzf3jbbsqv3m0000gp/T/tmpoo0qhdqt             --surveytiles dark-tiles.txt              --footprint tiles.fits              --positioners ./tmp_ETC_pos.txt             --starmask 60129542144             --rundate 2018-06-21
rewriting tilefile for tileid 10000
```

**Comment by moustakas** [comment]
I think I figured out the problem -- `bin/fiberassign` assumes/hard-codes a 5-digit tileid (see, e.g., [here](https://github.com/desihub/fiberassign/blob/master/bin/fiberassign#L158), [here](https://github.com/desihub/fiberassign/blob/master/bin/fiberassign#L166), and [here](https://github.com/desihub/fiberassign/blob/master/bin/fiberassign#L194)), whereas I'm using a tileid that starts at 100000, and so the last digit is being clipped.

**Comment by moustakas** [comment]
Fixed in #126.
